### PR TITLE
Add prefix contain Map or List properties

### PR DESCRIPTION
**Pull request by hui1110**
As title.
When the property prefix contains Map or list, the property prefix is incomplete, they are not in the [appendix ](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/appendix.html##configuration-properties)of the reference document. Manually add the full prefix of the property to ensure that all properties are in the [appendix ](https://microsoft.github.io/spring-cloud-azure/4.0.0-beta.4/4.0.0-beta.4/reference/html/appendix.html##configuration-properties)of the reference document.

### CODE REVIEW

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 1]
```diff
@@ -983,6 +983,482 @@
...
+      "name": "spring.cloud.azure.appconfiguration.client.headers[0].values",
+      "type": "java.util.List<java.lang.String>",
+      "description": "List of values of the header.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.implementation.properties.core.client.HttpClientConfigurationProperties"
+    },
+    {
+      "name": "spring.cloud.azure.client.http.headers[0].name",
+      "type": "java.lang.String",
+      "description": "The name of the header.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.context.AzureGlobalProperties$GlobalHttpClientConfigurationProperties"
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:1003]: Same here.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 2]
```diff
@@ -983,6 +983,482 @@
...
+    },
+    {
+      "name": "spring.cloud.azure.client.http.headers[0].values",
+      "type": "java.util.List<java.lang.String>",
+      "description": "List of values of the header.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.context.AzureGlobalProperties$GlobalHttpClientConfigurationProperties"
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:1009]: Here.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 3]
```diff
@@ -983,6 +983,482 @@
...
+    },
+    {
+      "name": "spring.cloud.azure.eventhubs.processor.checkpoint-store.client.headers[0].name",
+      "type": "java.lang.String",
+      "description": "The name of the header.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.implementation.properties.core.client.HttpClientConfigurationProperties"
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:1015]: Here.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 4]
```diff
@@ -983,6 +983,482 @@
...
+    },
+    {
+      "name": "spring.cloud.azure.eventhubs.processor.checkpoint-store.client.headers[0].values",
+      "type": "java.util.List<java.lang.String>",
+      "description": "List of values of the header.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.implementation.properties.core.client.HttpClientConfigurationProperties"
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:1021]: Here.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 5]
```diff
@@ -983,6 +983,482 @@
...
+    },
+    {
+      "name": "spring.cloud.azure.keyvault.certificate.client.headers[0].name",
+      "type": "java.lang.String",
+      "description": "The name of the header.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.implementation.properties.core.client.HttpClientConfigurationProperties"
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:1027]: Here.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 6]
```diff
@@ -983,6 +983,482 @@
...
+    },
+    {
+      "name": "spring.cloud.azure.keyvault.certificate.client.headers[0].values",
+      "type": "java.util.List<java.lang.String>",
+      "description": "List of values of the header.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.implementation.properties.core.client.HttpClientConfigurationProperties"
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:1033]: Here.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 7]
```diff
@@ -983,6 +983,482 @@
...
+    },
+    {
+      "name": "spring.cloud.azure.keyvault.secret.client.headers[0].name",
+      "type": "java.lang.String",
+      "description": "The name of the header.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.implementation.properties.core.client.HttpClientConfigurationProperties"
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:1039]: Here.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 8]
```diff
@@ -983,6 +983,482 @@
...
+    },
+    {
+      "name": "spring.cloud.azure.keyvault.secret.client.headers[0].values",
+      "type": "java.util.List<java.lang.String>",
+      "description": "List of values of the header.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.implementation.properties.core.client.HttpClientConfigurationProperties"
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:1045]: Here.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 9]
```diff
@@ -983,6 +983,482 @@
...
+    },
+    {
+      "name": "spring.cloud.azure.storage.blob.client.headers[0].name",
+      "type": "java.lang.String",
+      "description": "The name of the header.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.implementation.properties.core.client.HttpClientConfigurationProperties"
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:1051]: Here.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 10]
```diff
@@ -983,6 +983,482 @@
...
+    },
+    {
+      "name": "spring.cloud.azure.storage.blob.client.headers[0].values",
+      "type": "java.util.List<java.lang.String>",
+      "description": "List of values of the header.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.implementation.properties.core.client.HttpClientConfigurationProperties"
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:1057]: Here.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 11]
```diff
@@ -983,6 +983,482 @@
...
+    },
+    {
+      "name": "spring.cloud.azure.storage.fileshare.client.headers[0].name",
+      "type": "java.lang.String",
+      "description": "The name of the header.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.implementation.properties.core.client.HttpClientConfigurationProperties"
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:1063]: Here.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 12]
```diff
@@ -983,6 +983,482 @@
...
+    },
+    {
+      "name": "spring.cloud.azure.storage.fileshare.client.headers[0].values",
+      "type": "java.util.List<java.lang.String>",
+      "description": "List of values of the header.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.implementation.properties.core.client.HttpClientConfigurationProperties"
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:1069]: Here.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 13]
```diff
@@ -983,6 +983,482 @@
...
+    },
+    {
+      "name": "spring.cloud.azure.storage.queue.client.headers[0].name",
+      "type": "java.lang.String",
+      "description": "The name of the header.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.implementation.properties.core.client.HttpClientConfigurationProperties"
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:1075]: Here.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 14]
```diff
@@ -983,6 +983,482 @@
...
+    },
+    {
+      "name": "spring.cloud.azure.storage.queue.client.headers[0].values",
+      "type": "java.util.List<java.lang.String>",
+      "description": "List of values of the header.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.implementation.properties.core.client.HttpClientConfigurationProperties"
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:1081]: Here.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 15]
```diff
@@ -983,6 +983,482 @@
...
+      "name": "spring.cloud.azure.keyvault.secret.property-sources[0].case-sensitive",
+      "type": "java.lang.Boolean",
+      "description": "Defines the constant for the property that enables/disables case-sensitive keys.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.implementation.keyvault.secrets.properties.AzureKeyVaultPropertySourceProperties"
+    },
+    {
+      "name": "spring.cloud.azure.keyvault.secret.property-sources[0].client.application-id",
+      "type": "java.lang.String",
+      "description": "Represents current application and is used for telemetry\/monitoring purposes.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.implementation.properties.core.client.HttpClientConfigurationProperties"
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:1093]: It should be `com.azure.spring.cloud.autoconfigure.properties.core.client.ClientConfigurationProperties`

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 16]
```diff
@@ -983,6 +983,482 @@
...
+      "name": "spring.cloud.azure.keyvault.secret.property-sources[0].client.connection-idle-timeout",
+      "type": "java.time.Duration",
+      "description": "Amount of time before an idle connection.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.implementation.properties.core.client.HttpClientConfigurationProperties"
+    },
+    {
+      "name": "spring.cloud.azure.keyvault.secret.property-sources[0].client.headers[0].name",
+      "type": "java.lang.String",
+      "description": "The name of the header.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.implementation.properties.core.client.HttpClientConfigurationProperties"
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:1111]: Here.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 17]
```diff
@@ -983,6 +983,482 @@
...
+    },
+    {
+      "name": "spring.cloud.azure.keyvault.secret.property-sources[0].client.headers[0].values",
+      "type": "java.util.List<java.lang.String>",
+      "description": "List of values of the header.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.implementation.properties.core.client.HttpClientConfigurationProperties"
```
Comment (chenrujun) [inline sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:1117]: Here.

**Review by chenrujun — CHANGES_REQUESTED** [review]

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json (hunk 10 above)
Comment (hui1110) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json]: updated

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 18]
```diff
@@ -983,6 +983,482 @@
     {
       "name": "spring.jms.servicebus.pool.use-anonymous-producers",
       "description": "Whether to use only one anonymous 'MessageProducer' instance. Set it to false to create one 'MessageProducer' every time one is required."
+    },
+    {
+      "name": "spring.cloud.azure.appconfiguration.client.headers[0].name",
+      "type": "java.lang.String",
+      "description": "The name of the header.",
+      "sourceType": "com.azure.spring.cloud.autoconfigure.implementation.properties.core.client.HttpClientConfigurationProperties"
```
[the remark this answers is not in the document]
Comment (hui1110) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json]: updated

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json (hunk 15 above)
Comment (hui1110) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json]: updated

**Review by chenrujun — APPROVED** [review]
LGTM.

File: sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json [hunk 19]
```diff
@@ -983,6 +983,482 @@
     {
       "name": "spring.jms.servicebus.pool.use-anonymous-producers",
       "description": "Whether to use only one anonymous 'MessageProducer' instance. Set it to false to create one 'MessageProducer' every time one is required."
+    },
+    {
+      "name": "spring.cloud.azure.appconfiguration.client.headers[0].name",
```
[the remark this answers is not in the document]
Comment (saragluna) [reply sdk/spring/spring-cloud-azure-autoconfigure/src/main/resources/META-INF/additional-spring-configuration-metadata.json:988]: yes, @hui1110 please help fix it too.

**Pull request merged**